buildpack/telemetry/dynatrace.py

```diff
@@ -1,110 +1,31 @@
+"""
 import logging
 import os
-import shutil
-import json
+from urllib.parse import urljoin
 
-from buildpack import util
+INGEST_ENDPOINT = "/api/v2/metrics/ingest"
 
-default_env = {
-    "DT_TENANTTOKEN": None,  # optional, default to one found in manifest.json
-    "DT_APPLICATIONID": None,  # optional, default not set
-    "DT_TENANT": None,  # required, environment ID (uuid format)
-    # "DT_PAAS_TOKEN": None,  # required
-    # "DT_SAAS_URL": None,  # required
-    "DT_LOGSTREAM": "stdout",
-    "DT_NETWORK_ZONE": None,  # optional, not sure what this means :D
-    "DT_CUSTOM_PROP": None,  # optional metadata e.g. Department=Acceptance Stage=Sprint
-    "DT_TAGS": None,  # optional tags e.g. MikesStuff easyTravel=Mike
-}
 
-
-def stage(buildpack_dir, build_path, cache_path):
-    if is_enabled():
-        try:
-            util.resolve_dependency(
-                "dynatrace.agent",
-                build_path,  # DOT_LOCAL_LOCATION,
-                buildpack_dir=buildpack_dir,
-                cache_dir=cache_path,  # CACHE_DIR,
-                unpack=True,
-                overrides={
-                    "url": os.environ.get("DT_SAAS_URL"),
-                    "environment": os.environ.get("DT_TENANT"),
-                    "token": os.environ.get("DT_PAAS_TOKEN"),
-                },
-                ignore_cache=True
-            )
-        except Exception as e:
-            logging.warning(
-                "Dynatrace agent download and unpack failed", exc_info=True
-            )
+def is_enabled():
+    return "DT_PAAS_TOKEN" in os.environ.keys() and \
+           "DT_SAAS_URL" in os.environ.keys()
 
 
-def update_config(m2ee, app_name):
+def token():
     if not is_enabled():
-        logging.debug(
-            "Skipping Dynatrace setup, no DT_TENANTTOKEN found in environment"
-        )
-        return
-    logging.info("Enabling Dynatrace")
-    try:
-        manifest = get_manifest()
-    except Exception as e:
-        logging.warning(
-            "Failed to parse Dynatrace manifest file", exc_info=True
-        )
-        return
-
-    # dynamic default
-    default_env.update({"DT_TENANTTOKEN": manifest.get("tenantToken")})
-
-    for key, dv in default_env.items():
-        value = os.environ.get(key, dv)
-        if value:
-            util.upsert_custom_environment_variable(m2ee, key, value)
-    util.upsert_custom_environment_variable(
-        m2ee, "DT_CONNECTION_POINT", get_connection_endpoint()
-    )
-
-    agent_path = os.path.join(".local", get_agent_path())
-    if not os.path.exists(agent_path):
-        raise Exception(
-            "Dynatrace Agent not found: {agent_path}".format(agent_path)
-        )
-
-    util.upsert_javaopts(
-        m2ee,
-        [
-            "-agentpath:{path}".format(path=os.path.abspath(agent_path)),
-            "-Xshare:off",
-        ],
-    )
+        logging.debug("Dynatrace is not enabled, cannot find DT_PAAS_TOKEN")
```

Comment:
This is a bit misleading as `is_enabled()` checks for two parameters. Also we are fetching from environment vars in two separate places. Would be nice to consolidate the vars and checks.

Reply:
Consolidated fetching of two parameters. Added info log to state Dynatrace integration is configured. Still keeping `is_enabled()` separate because it has multiple usages and wanna keep it without logging.